src/makeCalculator.js

```diff
@@ -37,7 +37,47 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  return {
+    result: 0,
+
+    add: function(num) {
+      this.result += num;
+
+      return this;
+    },
+
+    subtract: function(num) {
+      this.result -= num;
+
+      return this;
+    },
+
+    multiply: function(num) {
+      this.result *= num;
+
+      return this;
+    },
+
+    divide: function(num) {
+      if (num !== 0) {
+        this.result /= num;
+      }
+
+      return this;
+    },
+
+    reset: function() {
+      this.result = 0;
+
+      return this;
+    },
+
+    operate: function(callback, num) {
+      callback.call(this, num);
```

Comment:
In the operate function, there's no checking if the callback is a function. This could lead to a runtime error if a non-function is passed. Please add an appropriate check.